odbc/src/api/diag.rs

```diff
@@ -150,41 +106,25 @@ pub unsafe fn get_diag_field(
                 match diag_identifier {
                     // NOTE: return code is handled by driver manager; just return success
                     DiagType::SQL_DIAG_RETURNCODE => SqlReturn::SUCCESS,
-                    DiagType::SQL_DIAG_SQLSTATE => match is_wstring {
-                        true => i16_len::set_output_wstring_as_bytes(
-                            error.get_sql_state(),
-                            diag_info_ptr,
-                            buffer_length as usize,
-                            string_length_ptr,
-                        ),
-                        false => i16_len::set_output_string(
-                            error.get_sql_state(),
-                            diag_info_ptr as *mut u8,
-                            buffer_length as usize,
-                            string_length_ptr,
-                        ),
-                    },
+                    DiagType::SQL_DIAG_SQLSTATE => i16_len::set_output_wstring_as_bytes(
+                        error.get_sql_state(),
+                        diag_info_ptr,
+                        buffer_length as usize,
+                        string_length_ptr,
+                    ),
```

Comment:
Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

Reply:
> Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

@terakilobyte Please do :)